src/conn.c

```diff
@@ -2044,19 +2049,29 @@ static void
 _readLoop(void  *arg)
 {
     natsStatus  s = NATS_OK;
-    char        buffer[DEFAULT_BUF_SIZE];
+    char        *buffer;
     natsSock    fd;
     int         n;
 
     natsConnection *nc = (natsConnection*) arg;
 
     natsConn_Lock(nc);
 
+    fd = nc->sockCtx.fd;
+
+    buffer = NATS_MALLOC(nc->opts->ioBufSize);
```

Comment:
Maybe we could capture the option value in a local variable here so that we don't need to de-reference nc->opts in the natsSock_Read() call down there.

Reply:
See 9c8ad96.